salt/orchestrate/reddit/init.sls

```diff
@@ -0,0 +1,109 @@
+    - name: /etc/salt/cloud.maps.d/{{ VPC_RESOURCE_SUFFIX }}_{{ app_name }}_map.yml
+    - source: salt://orchestrate/aws/map_templates/instance_map.yml
+    - template: jinja
+    - makedirs: True
+    - context:
+        business_unit: {{ BUSINESS_UNIT }}
+        environment_name: {{ ENVIRONMENT }}
+        purpose_prefix: {{ PURPOSE_PREFIX }}
+        securitygroupids:
+          webapp: {{ salt.boto_secgroup.get_group_id(
+              'webapp-{}'.format(ENVIRONMENT), vpc_name=VPC_NAME) }}
+          default: {{ salt.boto_secgroup.get_group_id(
+              'default', vpc_name=VPC_NAME) }}
+          salt-master: {{ salt.boto_secgroup.get_group_id(
+            'salt_master-{}'.format(ENVIRONMENT), vpc_name=VPC_NAME) }}
+          consul-agent: {{ salt.boto_secgroup.get_group_id(
+            'consul-agent-{}'.format(ENVIRONMENT), vpc_name=VPC_NAME) }}
+        subnetids: {{ subnet_ids }}
+        tags:
+          Department: {{ BUSINESS_UNIT }}
+          OU: {{ BUSINESS_UNIT }}
+          Environment: {{ ENVIRONMENT }}
+    - require:
+        - file: load_{{ app_name }}_cloud_profile
+
+ensure_instance_profile_exists_for_{{ app_name }}:
+  boto_iam_role.present:
+    - name: {{ app_name }}-instance-role
+
+deploy_{{ app_name }}_cloud_map:
+  salt.function:
+    - tgt: 'roles:master'
+    - tgt_type: grain
+    - name: saltutil.runner
+    - arg:
+        - cloud.map_run
+    - kwarg:
+        path: /etc/salt/cloud.maps.d/{{ VPC_RESOURCE_SUFFIX }}_{{ app_name }}_map.yml
+        parallel: True
+        full_return: True
+    - require:
+        - file: generate_{{ app_name }}_cloud_map_file
+
+load_pillar_data_on_{{ app_name }}_nodes:
+  salt.function:
+    - name: saltutil.refresh_pillar
+    - tgt: 'P@roles:{{ app_name }} and G@environment:{{ ENVIRONMENT }}'
+    - tgt_type: compound
+    - require:
+        - salt: deploy_{{ app_name }}_cloud_map
+
+populate_mine_with_{{ app_name }}_node_data:
+  salt.function:
+    - name: mine.update
+    - tgt: 'P@roles:{{ app_name }} and G@environment:{{ ENVIRONMENT }}'
+    - tgt_type: compound
+    - require:
+        - salt: load_pillar_data_on_{{ app_name }}_nodes
+
+deploy_consul_agent_to_{{ app_name }}_nodes:
+  salt.state:
+    - tgt: 'P@roles:{{ app_name }} and G@environment:{{ ENVIRONMENT }}'
+    - tgt_type: compound
+    - sls:
+        - consul
+        - consul.dns_proxy
+
+restart_consul_service_to_load_updated_configs:
+  salt.function:
+    - tgt: 'P@roles:{{ app_name }} and G@environment:{{ ENVIRONMENT }}'
+    - tgt_type: compound
+    - name: service.restart
+    - arg:
+        - consul
+    - require:
+        - salt: deploy_consul_agent_to_{{ app_name }}_nodes
+
+build_{{ app_name }}_nodes:
+  salt.state:
+    - tgt: 'P@roles:{{ app_name }} and G@environment:{{ ENVIRONMENT }}'
+    - tgt_type: compound
+    - highstate: True
+    - require:
+        - salt: deploy_consul_agent_to_{{ app_name }}_nodes
+        - salt: restart_consul_service_to_load_updated_configs
+
```

Comment:
This is just missing execution of the post_deploy state from the reddit formula. You'll just want to run that using the `subset: 1` parameter so that we don't run into race conditions with multiple hosts trying to provision the same values on the databases.